PathPlanning/Bug_Planning/bug.py

```diff
@@ -0,0 +1,296 @@
+"""
+Bug Planning
+author: Sarim Mehdi(muhammadsarim.mehdi@studio.unibo.it)
+Source: https://sites.google.com/site/ece452bugalgorithms/
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+class Bug():
+    def __init__(self, startx, starty, goalx, goaly, obsx, obsy):
+        self.goalx = goalx
+        self.goaly = goaly
+        self.obsx = obsx
+        self.obsy = obsy
+        self.rx = [startx]
+        self.ry = [starty]
+        self.outx = []
+        self.outy = []
+        for o_x, o_y in zip(obsx, obsy):
+            for addx, addy in zip([1, 0, -1, -1, -1, 0, 1, 1],
+                                  [1, 1, 1, 0, -1, -1, -1, 0]):
+                candx, candy = o_x+addx, o_y+addy
+                valid_point = True
+                for _x, _y in zip(obsx, obsy):
+                    if candx == _x and candy == _y:
+                        valid_point = False
+                        break
+                if valid_point:
+                    self.outx.append(candx), self.outy.append(candy)
+
+    def mov_normal(self):
+        return self.rx[-1] + np.sign(self.goalx - self.rx[-1]), \
+               self.ry[-1] + np.sign(self.goaly - self.ry[-1])
+
+    def mov_to_next_obs(self, visitedx, visitedy):
+        for addx, addy in zip([1, 0, -1, 0], [0, 1, 0, -1]):
+            cx, cy = self.rx[-1] + addx, self.ry[-1] + addy
+            for _x, _y in zip(self.outx, self.outy):
+                use_pt = True
+                if cx == _x and cy == _y:
+                    for v_x, v_y in zip(visitedx, visitedy):
+                        if cx == v_x and cy == v_y:
+                            use_pt = False
+                            break
+                    if use_pt:
+                        return cx, cy, False
+                if not use_pt:
+                    break
+        return self.rx[-1], self.ry[-1], True
+
+    def bug0(self):
+        mov_dir = 'normal'
+        candx, candy = -9999, -9999
+        plt.plot(self.obsx, self.obsy, ".k")
```

Comment:
Could you please plotting function calls are covered by `show_animation` like:
https://github.com/AtsushiSakai/PythonRobotics/blob/fa227083372445f0f440b5d5ad878d9201ed7476/PathPlanning/AStar/a_star.py#L87-L93